server/redeemer.go

```diff
@@ -0,0 +1,383 @@
+	// Here we can configure things like TLS
+	opts := []grpc.ServerOption{}
+	// var s *grpc.Server
+	s := grpc.NewServer(opts...)
+	r.server = s
+
+	net.RegisterTicketRedeemerServer(s, r)
+
+	return s.Serve(listener)
+}
+
+// Stop stops the Redeemer server
+func (r *Redeemer) Stop() {
+	close(r.quit)
+	r.server.Stop()
+}
+
+// QueueTicket adds a ticket to the ticket queue
+func (r *Redeemer) QueueTicket(ctx context.Context, ticket *net.Ticket) (*net.QueueTicketRes, error) {
+	t := pmTicket(ticket)
+	if r.recipient != t.Recipient {
+		return nil, status.Error(codes.PermissionDenied, fmt.Sprintf("invalid ticket recipient 0x%x, expected %v", ticket.TicketParams.Recipient, r.recipient.Hex()))
+	}
+	if err := r.sm.QueueTicket(t); err != nil {
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+	glog.Infof("ticket queued sender=0x%x", ticket.Sender)
+
+	return &net.QueueTicketRes{}, nil
+}
+
+// MaxFloat is a unary RPC method to request the max float value for a sender
+func (r *Redeemer) MaxFloat(ctx context.Context, req *net.MaxFloatReq) (*net.MaxFloatUpdate, error) {
+	maxFloat, err := r.sm.MaxFloat(ethcommon.BytesToAddress(req.Sender))
+	if err != nil {
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+
+	return &net.MaxFloatUpdate{MaxFloat: maxFloat.Bytes()}, nil
+}
+
+// MonitorMaxFloat starts a server-side stream to the client to send max float updates for sender specified in the request
+func (r *Redeemer) MonitorMaxFloat(req *net.MaxFloatReq, stream net.TicketRedeemer_MonitorMaxFloatServer) error {
+	sender := ethcommon.BytesToAddress(req.Sender)
+
+	sink := make(chan struct{}, 10)
+	sub := r.sm.SubscribeMaxFloatChange(sender, sink)
+	defer sub.Unsubscribe()
+	for {
+		select {
+		case <-r.quit:
+			return nil
+		case err := <-sub.Err():
+			if err == nil {
+				return status.Error(codes.Canceled, "subscription closed")
+			}
+			return status.Error(codes.Canceled, err.Error())
+		case <-sink:
+			maxFloat, err := r.sm.MaxFloat(sender)
+			if err != nil {
+				glog.Error(err)
+			}
+			if err := stream.Send(&net.MaxFloatUpdate{MaxFloat: maxFloat.Bytes()}); err != nil {
+				glog.Error(err)
+				if err == io.EOF {
+					return err
+				}
+			}
+		}
+	}
+}
+
+type RedeemerClient struct {
+	conn *grpc.ClientConn
+	rpc  net.TicketRedeemerClient
+
+	senders map[ethcommon.Address]*remoteSender
+	mu      sync.RWMutex
+	quit    chan struct{}
+	sm      pm.SenderManager
+	tm      pm.TimeManager
+}
+
+type remoteSender struct {
+	maxFloat   *big.Int
+	stream     net.TicketRedeemer_MonitorMaxFloatClient
+	done       chan struct{}
+	lastAccess time.Time
+}
+
+// NewRedeemerClient instantiates a new client for the ticket redemption service
+// The client implements the pm.SenderMonitor interface
+func NewRedeemerClient(uri *url.URL, sm pm.SenderManager, tm pm.TimeManager) (*RedeemerClient, error) {
+	conn, err := grpc.Dial(
+		uri.String(),
+		grpc.WithBlock(),
+		grpc.WithTimeout(GRPCConnectTimeout),
+		grpc.WithInsecure(),
```

Comment:
Could we use `grpc.WithTransportCredentials(credential.NewTLS(tlsConfig))` like we do [here](https://github.com/livepeer/go-livepeer/blob/d0ea014d68488c1bc26c4e5882949a0d1dc1c86c/server/rpc.go#L225) so its easier to swap in a valid TLS config with `InsecureSkipVerify: false` in the future?

Reply:
I assume we no longer need `grpc.WithInsecure()` then because we now have valid transport credentials that accomplish the same?

Reply:
fixed in https://github.com/livepeer/go-livepeer/pull/1513/commits/a8e3d49938dcf3a62e11bff72accdc054c2d205d

Reply:
fixed in https://github.com/livepeer/go-livepeer/pull/1513/commits/6b7ac892cd933ab613b66a1cd5ae30cd990c128e

Reply:
Yeah the transport credentials should accomplish the same thing and we should be able to pass in valid credentials with `InsecureSkipVerify: false` in the future by updating `tlsConfig` passed in.